jupyter_core/paths.py

```diff
@@ -428,7 +432,7 @@ def secure_write(fname, binary=False):
         if os.name != 'nt':
             # Enforce that the file got the requested permissions before writing
             file_mode = get_file_mode(fname)
-            if 0o0600 != file_mode:
+            if 0o0600 != file_mode and not allow_insecure_writes:
```

Comment:
I was thinking about this (although I would have just logged the warning in the 'insecure' case, rather than unconditionally) but felt it would occur (essentially) at _every_ kernel startup request on applicable systems and could be lost in the general log messages (particularly when DEBUG is enabled, which I often recommend). However, I see that `warnings.warn()` just logs once, but again, that could be "lost in the noise".

I do like the idea of informing the admin that this setting is in effect, more or less confirming their intention.  I wonder if placing this warning message in `JupyterApp.initialize()`, or equivalent, might be more useful.  That way it's logged before any "violation" takes place.  We'd probably want a bold "WARNING" and not rely on the logging level indicator for something like this.  The downside would be for those apps that use `jupyter_core` but bypass `JupyterApp`.

This message would be something like...
`WARNING: Insecure writes have been enabled via environment variable  'JUPYTER_ALLOW_INSECURE_WRITES'!  If this is not intended, remove the variable or set its value to 'False'.`

Reply:
I'd be fine with that. Having the warning just-in-time is fine too imo. Just-in-time also catches when an application sets the environment after it loads before it runs something but that's probably not an edge case worth solving for on purpose. Up to you how you want to address it, i'd be good with merging with what's been described.

Reply:
I'd be fine with that. Having the warning just-in-time is fine too imo. Just-in-time also catches when an application sets the environment after it loads before it runs something but that's probably not an edge case worth solving for on purpose. Up to you how you want to address it, i'd be good with merging with what's been described.